src/test/java/com/tinkoff/edu/AppTest.java

```diff
@@ -5,48 +5,189 @@
-        Assertions.assertEquals(LoanResponseType.APPROVED, loanResponse.getIsAccepted());
-        Assertions.assertEquals(1, loanResponse.getRequestId());
-        finalCheck(LoanResponseType.APPROVED, 1, loanResponse);
+        finalCheck(LoanResponseType.DECLINED, loanResponse);
+    }
+
+    @Test
+    @DisplayName("Проверка одобрения заявки: loanType = IP")
+    public void checkApproveRequestForIP() {
+        loanRequest = new LoanRequest(12, 10000, LoanRequestType.IP);
+        repository = new VariableLoanCalcRepository();
+        controller = new LoanCalcController(new CalculationsWithAllParamsService(repository));
+        LoanResponse loanResponse = controller.createRequest(loanRequest);
+        System.out.println("Your request number is: " + loanResponse.getRequestId() + "; "
+                + "Your request status is: " + loanResponse.getIsAccepted());
+        finalCheck(LoanResponseType.DECLINED, loanResponse);
     }
 
     @Test
     @DisplayName("Проверка одобрения заявки с произвольным id")
     public void initCustomRequestId() {
+        loanRequest = new LoanRequest(10, 1000, LoanRequestType.IP);
         final int newId = 50;
         repository = new VariableLoanCalcRepository(newId);
         controller = new LoanCalcController(new CalculationsWithAllParamsService(repository));
         LoanResponse loanResponse = controller.createRequest(loanRequest);
         System.out.println("Your request number is: " + loanResponse.getRequestId() + "; "
                 + "Your request status is: " + loanResponse.getIsAccepted());
-        finalCheck(LoanResponseType.APPROVED, newId + 1, loanResponse);
+        finalCheck(LoanResponseType.DECLINED, newId + 1, loanResponse);
+    }
+
+    @Test
+    @DisplayName("Проверка выброса сключения для заявки = null")
+    public void checkNullRequest() {
+        repository = new VariableLoanCalcRepository();
+        controller = new LoanCalcController(new CalculationsWithAllParamsService(repository));
+        try {
+            controller.createRequest(null);
+            fail("No exception caught");
+        } catch (IllegalArgumentException exception) {
+        }
+    }
+
+    @Test
+    @DisplayName("Проверка выброса сключения для заявки с amount <= 0 ")
+    public void shouldGetErrorWhenApplyZeroOrNegativeAmountRequest() {
+        try {
+            loanRequest = new LoanRequest(12, 0, LoanRequestType.IP);
+            fail("No exception caught");
+        } catch (IllegalArgumentException exception) {
+        }
+    }
+
+    @Test
+    @DisplayName("Проверка выброса сключения для заявки с months <= 0 ")
+    public void shouldGetErrorWhenApplyZeroOrNegativeMonthsRequest() {
+        try {
+            loanRequest = new LoanRequest(-10, 10000, LoanRequestType.IP);
+            fail("No exception caught");
+        } catch (IllegalArgumentException exception) {
+        }
+    }
+
+    private void finalCheck(LoanResponseType expectedLoanType, LoanResponse loanResponse) {
```

Comment:
не стоит, лучше использовать нативный, под капот все равно практически никаких упрощений не заехало
```suggestion
    assertEquals(expectedLoanType, loanResponse.getIsAccepted());
```

Reply:
не понял к чему этот коммент и что мне надо поправить